### DISCUSSION TITLE
Questions about how the Shouldy NuGet package has been built

### DISCUSSION THREAD

**Discussion by PureKrome** [shouldly/shouldly#786] [Q&A] [answered]
G'Day! 👋🏻 

Got a question about the Shouldly nuget package. Currently it's
- Embedded (vs a pdb)
- ~Not sourcelinked~ (updated: [it is actually](https://github.com/shouldly/shouldly/blob/master/src/Shouldly/Shouldly.csproj#L27))
- Not optimized
- Not deterministic

I'm not suggesting these settings are wrong - I'm curious to know the decisions behind this.

example ref: [build.targets](https://github.com/shouldly/shouldly/blob/master/src/Shouldly/build.targets).

```
<Project>
  <PropertyGroup>
    <DebugSymbols>true</DebugSymbols>
    <Optimize>false</Optimize>
    <DebugType>embedded</DebugType>
    <Deterministic>false</Deterministic>
    <DeterministicSourcePaths>false</DeterministicSourcePaths>
  </PropertyGroup>
</Project>
```

I stumbled across this when I was upgrading a project from NET:shit: to NS20/NET50 and I couldn't generate any PDB's because I was taking a dependency on this project, which was forcing _my_ project to be 'embedded' DebugType.

cheers!

**Comment by SimonCropp** [comment]
@PureKrome so those settings are optimised for a test project consuming shouldy (the most common scenario). So in that case the pdb type/optimized/deterministic should not matter. 

I am guessing that you are not consuming shoudly from a test project but rather extending shoudly?

**Comment by SimonCropp** [comment, marked as answer]
Also: the reason those settings are required is so that shouldly can properly walk the stack trace, find the root method, and generate the correct error message

**Comment by PureKrome** [comment]
> I am guessing that you are not consuming shoudly from a test project but rather extending shoudly?

Nope - I have some old _library_ code I was experimenting with, updating from NET:poop: to NS20/NET50. It shouldn't be in my library, but I was trying to see if I could get an easy upgrade/quick win. (proper solution: remove it from library and just keep using them in tests).

I couldn't `dotnet pack` my library because I needed pdb's, and they were not getting generated. My test projects don't need pdb's and don't need to get packed, which was why i've not noticed this before.

So I was just curious to understanding what was going on.

I guess it's been answered 👍🏻
